docs/pages/api-docs/data-grid/grid-cell-params.md

```diff
@@ -21,7 +21,7 @@ import { GridCellParams } from '@mui/x-data-grid';
 | <span class="prop-name">getValue</span>                                                      | <span class="prop-type">(id: GridRowId, field: string) =&gt; GridCellValue</span> | Get the cell value of a row and field.                           |
 | <span class="prop-name">hasFocus</span>                                                      | <span class="prop-type">boolean</span>                                            | If true, the cell is the active element.                         |
 | <span class="prop-name">id</span>                                                            | <span class="prop-type">GridRowId</span>                                          | The grid row id.                                                 |
-| <span class="prop-name optional">isEditable<sup><abbr title="optional">?</abbr></sup></span> | <span class="prop-type">boolean</span>                                            | If true, the cell is editable.                                   |
+| <span class="prop-name optional">isEditable<sup><abbr title="optional">?</abbr></sup></span> | <span class="prop-type">boolean \| undefined</span>                               | If true, the cell is editable.                                   |
```

Comment:
There's already the question mark to sign that the property is optional. Can `undefined` be removed?

![image](https://user-images.githubusercontent.com/42154031/145146424-e4b768ec-3e98-4caf-b5c2-716db85d6251.png)

Reply:
As told in the description of the PR, it is the `typeToString` of TS which puts the `| undefined`.
Which is not wrong, but kind of heavy.
They have options on that method but I did not find one to remove it.

Reply:
Have you tried to use `node.getText`?

```diff
diff --git a/docs/scripts/api/buildInterfacesDocumentation.ts b/docs/scripts/api/buildInterfacesDocumentation.ts
index c7257f46a..7401dd4b3 100644
--- a/docs/scripts/api/buildInterfacesDocumentation.ts
+++ b/docs/scripts/api/buildInterfacesDocumentation.ts
@@ -51,11 +51,10 @@ const parseProperty = (propertySymbol: ts.Symbol, project: Project): ParsedPrope
   tags: getSymbolJSDocTags(propertySymbol),
   symbol: propertySymbol,
   isOptional: !!propertySymbol.declarations?.find(ts.isPropertySignature)?.questionToken,
-  typeStr: project.checker.typeToString(
-    project.checker.getTypeOfSymbolAtLocation(propertySymbol, propertySymbol.valueDeclaration!),
-    propertySymbol.valueDeclaration,
-    ts.TypeFormatFlags.NoTruncation,
-  ),
+  typeStr:
+    propertySymbol.valueDeclaration && ts.isPropertySignature(propertySymbol.valueDeclaration)
+      ? propertySymbol.valueDeclaration.type?.getText() || ''
+      : '',
 });
 
 function generateMarkdownFromProperties(
```

Reply:
https://github.com/mui-org/material-ui-x/pull/3320/commits/066f0c4daa216905786edaf9b1c7123037ab6956

Here is an example that replace `typeToString` with `getText` when available.